hell/adventcalendar/2024/16/index.md

```diff
@@ -1,52 +1,74 @@
 
+Whether the `isOpen` class is applied or not, a screenreader would announce this button as **"Click me!, button"**.
 
 ```html
-<!DOCTYPE html>
-<html lang="en">
-  <head>
-	  <meta charset="utf-8" />
-	  <title>Webpage starter</title>
-  </head>
-  <body>
-    <h1>Hello World!</h1>
-  </body>
-</html>
+<button>Click me!</button>
+<p class="isOpen">Content that displays depending on a conditionally applied class.</p>
 ```
 
-## Lorem ipsum dolor sit amet.
+One of the principles of web accessibility is to be perceivable. That means not relying exclusively on one sense, like sight, to provide information. In the above case, if I can’t see the content being shown or hidden, how do I know my button click did anything?
 
-* Lorem, ipsum.
-* Lorem, ipsum.
-* Lorem, ipsum.
-* Lorem, ipsum.
+We *need* a corresponding semantic update to indicate what happened. This is what the `aria-expanded` attribute provides.
 
-![image](NSvIE.jpg)
+The following button would read: **"Click me!, expanded, button"**.
 
-or
+```html
+<button aria-expanded="true">Click me!</button>
+<p>Content that displays depending on button's expanded state.</p>
+```
+
+This one would be **"Click me!, collapsed, button"**.
+
+```html
+<button aria-expanded="false">Click me!</button>
+<p>Content that displays depending on button's expanded state.</p>
+```
+
+The `aria-expanded` attribute also provides a ready-made CSS selector for collapsed and expanded states. So, we don’t need to add an `isOpen` class at all.
+
+If you’ve used the `:checked` pseudo-class before, you’ll be familiar with the idea of styling scoped to a particular state. It’s a good idea. As a bonus, you get a mini accessibility test in your CSS! The styles only apply if the semantics are correct.
+
+```css
+button[aria-expanded="false"] > * {
+ /* collapsed content styles here */
+ display: none;
+}
+
+button[aria-expanded="true"] > * {
+ /* expanded content styles here */
+ display: block;
+}
+```
 
-<img src="NSvIE.jpg" width="800" height="161" loading="lazy">
+I first learned of this technique through [Heydon Pickering’s Inclusive Components](https://inclusive-components.design/collapsible-sections/) and have been using it ever since. It’s simple, effective, and a great way to keep accessibility in mind at every stage of development.
 
-Lorem ipsum dolor, sit amet consectetur adipisicing elit. Ipsum, eligendi nobis impedit perferendis, saepe neque ullam laborum laudantium [recusandae]() expedita distinctio. At accusamus necessitatibus mollitia, quas officia incidunt nisi corrupti.
\ No newline at end of file
+<p class="highlight">
+<strong>Note:</strong> You will not need <code>aria-expanded</code> on the HTML <code>details</code> element. Stateful information is automatically provided through its <code>open</code> attribute. <a href="https://www.scottohara.me/blog/2022/09/12/details-summary.html">See Scott O’Hara for more details on <code>details</code></a>.
```

Comment:
Similarly, unsure whether it's worth adding this note, but wanted to clarify ARIA is only needed with custom disclosure widgets, not the native HTML one

Reply:
I like it! Readers may ask themselves that question!